cpp/tests/io/text/data_chunk_source_test.cpp

```diff
@@ -342,4 +342,33 @@ TEST_F(DataChunkSourceTest, BgzipCompressedSourceVirtualOffsets)
   test_source(input, *source);
 }
 
+TEST_F(DataChunkSourceTest, BgzipSourceVirtualOffsetsSingleCompressedGZipBlock)
+{
+  auto const filename = temp_env->get_temp_filepath("bgzip_source_offsets_single_compressed_block");
+  std::string const input{"collection unit brings"};
+  std::string head_garbage{"garbage"};
+  // make the head_garbage large enough so the local offset will overrun the compressed block size
+  for (int i = 0; i < 10; i++) {
+    head_garbage += head_garbage;
+  }
```

Comment:
1. Just write a long string, don't construct it. clang-format may have opinions on wrapping / line length, I'm not sure.
2. How big does it need to be? Does it need to be 11 * 7 = 77 characters? (Replace 77 with the correct size below)
```suggestion
  // Make the head_garbage 77 characters so the local offset will overrun the compressed block size
  std::string head_garbage{"garbagegarbagegarbagegarbagegarbagegarbagegarbagegarbagegarbagegarbagegarbage"};
```

Reply:
he is doubling everytime in the loop. 7*2^10.

Reply:
I need a long string that is easily compressible to make sure that my local offset is much larger than the compressed block size

Reply:
You create a large string without a for-loop using
```
auto large_enough_size = 7* (1<<10);
std::string head_garbage( large_enough_size, 'G' );
```
The `head_garbage` will contain `large_enough_size` 'G' characters.
Of course this will compress better than `garbagegarbagegarbage...` in case that matters.

Reply:
> it is doubling everytime in the loop. 7*2^10.

Of course. Apologies for the silly mistake. 😅

Reply:
I like @davidwendt’s solution above, if that solves the problem.

Reply:
good point, for the garbage, it doesn't matter what the contents look like, as long as it's not valid BGZIP. I changed the same in other locations. For the input in the other tests, I actually prefer this pattern, since it creates output that makes off-by-one errors or similar in the offsets visible.